Fixes from PR#89 (and initially from PR#59) (also fixes #58)

johlju opened the pull request:
This is the new PR (again) from the PR #89 (and initially from PR #59)

_This is from the work by @Matticusau in the closed PR #59 and closed PR #23, which @aultt continued the work on in PR #89. Now @johlju is continuing the work to get this merged._

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xsqlserver/114)
<!-- Reviewable:end -->

msftclas commented:
Hi __@johlju__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

johlju commented:
**Wait for review on this one,** I will go thru all the comments in #89 and fix those first. I let you know when it's ready for review. Just wanted to see that I got all files in.

johlju commented:
**This one is ready for review now.** Resolved all the comments from the review of PR #89.

johlju commented:
Maintainers have a lot of work, so until they come around maybe someone in the community could start the review on this?

johlju commented:
@mbreakey3 When you have time, could you review this PR?

johlju commented:
I squashed (concatenated) a lot of commits.

mbreakey3 commented:
Reviewed 5 of 8 files at r3.
Review status: 5 of 8 files reviewed at latest revision, 23 unresolved discussions.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 28 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUssa3Sz4yaxcycRGs:-KRUssa3Sz4yaxcycRGt:b-whxubn) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L28)):*
> ```PowerShell
>         Protocol = [System.String]
>         ServerName = [System.String]
>         TCPPort = [System.Int32]
> ```

TcpPort

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 35 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUt-XobMbE8LRe03UG:-KRUt-XobMbE8LRe03UH:b-sewwi4) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L35)):*
> ```PowerShell
>     if ($null -ne (Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\MSSQLServer\Client\ConnectTo' -Name $Name -ErrorAction SilentlyContinue))
>     {
>         $itemValue = Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\MSSQLServer\Client\ConnectTo' -Name $Name -ErrorAction SilentlyContinue
> ```

You shouldn't need to call this twice - just put this line above the if statement and then check that ($null -ne $itemValue)

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 39 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUwmORmQh81mU9ycil:-KRUwmORmQh81mU9ycim:bk9e3ub) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L39)):*
> ```PowerShell
>         $returnValue.Name = $Name
>         $itemConfig = $itemValue."$Name" -split ','
>         if ($itemConfig[0] -eq 'DBMSSOCN')
> ```

$itemConfig needs to be checked that it's non-null before indexing into it

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 42 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV5Wzvx14hS_t1xEKI:-KRV5Wzvx14hS_t1xEKJ:btbme7g) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L42)):*
> ```PowerShell
>         {
>             $returnValue.Protocol = 'TCP'
>             $returnValue.ServerName = $itemConfig[1]
> ```

before indexing into the area - $itemConfig needs to be checked to ensure it has at least that many items

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 76 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUu0w8XeeOnHtm6JhA:-KRUu0w8XeeOnHtm6JhB:b-d0kccz) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L76)):*
> ```PowerShell
> 
>         [System.Int32]
>         $TCPPort,
> ```

TcpPort

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 105 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUuSxZ8_6ejlwJ8T70:-KRUuSxZ8_6ejlwJ8T71:bxemln4) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L105)):*
> ```PowerShell
>         if ($null -ne $currentValue -and $itemValue -ne $currentValue)
>         {
>             if ($PSCmdlet.ShouldProcess($Name,"Changing the client alias (64-bit)"))
> ```

single quotes around 'Changing...'

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 113 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUyI8A5OxXDYg5e0uy:-KRUyI8Bb5C9jjXzmaO2:b-v00odd) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L113)):*
> ```PowerShell
>         elseif ($null -eq $currentValue)
>         {
>             if ($PSCmdlet.ShouldProcess($Name,"Create client alias (64-bit)"))
> ```

single quotes around 'Create...'

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 135 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUybsw2N59cbIgtZg1:-KRUybsw2N59cbIgtZg2:b-momccj) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L135)):*
> ```PowerShell
>             if ($null -ne $currentValue -and $itemValue -ne $currentValue)
>             {
>                 if ($PSCmdlet.ShouldProcess($Name,"Changing the client alias (32-bit)"))
> ```

all of these should be single quotes

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 234 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUzUyQnKreRzZxaMKU:-KRUzUyQnKreRzZxaMKV:b6gk2k9) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L234)):*
> ```PowerShell
>                     Write-Debug -Message 'Named Pipes'
> 
>                     if ($itemConfig[0] -ne 'DBNMPNTW')
> ```

again, check to make sure this isn't null first

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 239 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUzapSnZN6xbIBTmQO:-KRUzapSnZN6xbIBTmQP:b-n9kef7) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L239)):*
> ```PowerShell
>                     }
> 
>                     if ($itemConfig[1] -ne "\\$ServerName\PIPE\sql\query")
> ```

check this as well to ensure there's >1 item in the area before indexing into it

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 248 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUzmzpHiaMMBdqawE9:-KRUzmzpHiaMMBdqawEA:b2rbo5k) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L248)):*
> ```PowerShell
>                 {
>                     Write-Debug -Message 'TCP'
>                     if ($itemConfig[0] -ne 'DBMSSOCN')
> ```

check for not null

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 253 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-5JXL1Jfutn8WkMr:-KRV-5JXL1Jfutn8WkMs:b-533fpv) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L253)):*
> ```PowerShell
>                     }
> 
>                     if ($itemConfig[1] -ne $ServerName)
> ```

check number of items in array

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 258 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-A5MzugdGOYfTY4-:-KRV-A5MzugdGOYfTY40:b-533fpv) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L258)):*
> ```PowerShell
>                     }
> 
>                     if ($itemConfig[2] -ne $TCPPort)
> ```

check number of items in array

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 279 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-Fvj0ct7-8_CeIKU:-KRV-Fvj0ct7-8_CeIKV:b-6xoaza) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L279)):*
> ```PowerShell
>                             Write-Debug -Message 'Named Pipes'
> 
>                             if ($itemConfig[0] -ne 'DBNMPNTW')
> ```

see above comment

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 294 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-IDIbjyWTE8hg8M2:-KRV-IDIbjyWTE8hg8M3:b-6xoaza) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L294)):*
> ```PowerShell
>                             Write-Debug -Message 'TCP'
> 
>                             if ($itemConfig[0] -ne 'DBMSSOCN')
> ```

see above comment

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 25 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV1klqmsGtISXbDoyT:-KRV1klqmsGtISXbDoyU:b-g43yy9) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L25)):*
> ```PowerShell
> try
> {
>     #region Pester Test Initialization
> ```

delete these two lines since you don't have anything here

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 43 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV22P4MGQV-AraQ4SY:-KRV22P4MGQV-AraQ4SZ:b5t5f19) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L43)):*
> ```PowerShell
>             $SqlAlias.Protocol = 'TCP'    
>         }
>     }
> ```

Could you add another test for Get? One for NP probably

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 99 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV2uxc1cFAJaZp6G1L:-KRV2uxduvOyxA0WpaR_:buah0sk) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L99)):*
> ```PowerShell
>         }
> 
>         It 'Should call any Remove-ItemProperty exactly 2 times (1 for 32bit and 1 for 64 bit reg keys) when alias should be absent' {
> ```

can you re-word this It?

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 128 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV38rE7aRmePTL4ddU:-KRV38rE7aRmePTL4ddV:b-fdap0) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L128)):*
> ```PowerShell
>         } -ModuleName $script:DSCResourceName
> 
>         It 'Should return $true when Test is passed as Alias thats already set'{
> ```

'...is passed an Alias...'

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 132 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV3C7m1L3Sx9OLcLqz:-KRV3C7m1L3Sx9OLcLr-:b-6xoaza) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L132)):*
> ```PowerShell
>         }
> 
>         It 'Should return $false when Test is passed as Alias that is not set'{
> ```

see above comment

---

*[Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1, line 106 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV3y-PZJgsNINDnwd5:-KRV3y-PZJgsNINDnwd6:b3yf927) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1#L106)):*
> ```PowerShell
>         Mock -CommandName Connect-SQL -MockWith {
>             # build a custom object to return which is close to the real SMO object
>             $smoObj = [PSCustomObject]@{
> ```

space between ']' and '@'

---

*[Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1, line 199 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV4f3ymnPcHbczYQKe:-KRV4f3ymnPcHbczYQKf:bu38rh) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1#L199)):*
> ```PowerShell
>                     Name = 'Node01'
>                 }
>                 , [PSCustomObject]@{
> ```

commas should be on the previous line - also space betwen ] and @

---

*[Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1, line 262 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV51dBwQXaMOAaTBq2:-KRV51dBwQXaMOAaTBq3:b-nhisj) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1#L262)):*
> ```PowerShell
> 
>                 'Microsoft.SqlServer.Management.Smo.AvailabilityReplica' {
>                     $object = [PSCustomObject]@{
> ```

space between ] and @

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/114)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reworking the xSqlAlias resource and the pester test. Finished with the resource yesterday, now I'm working on improving the pester test.

johlju commented:
Reworked the entire xSqlAlias resource and it's entire test. Fixing the comments in the previous code would have made the validation logic even harder to read. I hope now I got a better base that will be easier to improve on when I get more review comments. :smile: 
The resource is fully tested and I also tried to make the test hit every aspect. Let's see what the reviewer find :)

@mbreakey3 when you have time, please continue the review.

---

Reviewed 4 of 4 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 28 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUssa3Sz4yaxcycRGs:-KRrdAEJEdZKkSR0pHYQ:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L28)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> TcpPort
</details>
Done.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 35 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUt-XobMbE8LRe03UG:-KRre4Ad_2-wV_5QpWTi:b-8wjosc) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L35)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> You shouldn't need to call this twice - just put this line above the if statement and then check that ($null -ne $itemValue)
</details>
Reworked the entire resource. Please review from the beginning again. 

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 39 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUwmORmQh81mU9ycil:-KRreDQnCO_QMn2kp2Os:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L39)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> $itemConfig needs to be checked that it's non-null before indexing into it
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 42 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV5Wzvx14hS_t1xEKI:-KRreF_4sPu3p7koh7G6:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L42)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> before indexing into the area - $itemConfig needs to be checked to ensure it has at least that many items
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 76 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUu0w8XeeOnHtm6JhA:-KRrfj1yB2zXbo1eiBaT:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L76)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> TcpPort
</details>
Done

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 105 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUuSxZ8_6ejlwJ8T70:-KRrfp4eMbghAvvKHyXm:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L105)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> single quotes around 'Changing...'
</details>
Done

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 113 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUyI8A5OxXDYg5e0uy:-KRrfxcDMQklEWoyls1j:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L113)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> single quotes around 'Create...'
</details>
Done

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 135 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUybsw2N59cbIgtZg1:-KRrfzAa4rjP9QE-NHYC:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L135)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> all of these should be single quotes
</details>
Done

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 234 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUzUyQnKreRzZxaMKU:-KRrg0AYVsQWGLZkmPug:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L234)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> again, check to make sure this isn't null first
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 239 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUzapSnZN6xbIBTmQO:-KRrg1FejYN2Rus9IANV:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L239)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> check this as well to ensure there's >1 item in the area before indexing into it
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 248 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRUzmzpHiaMMBdqawE9:-KRrg2FSyZclSSBtlPnz:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L248)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> check for not null
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 253 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-5JXL1Jfutn8WkMr:-KRrg2sDq_LzvyCvtCd0:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L253)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> check number of items in array
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 258 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-A5MzugdGOYfTY4-:-KRrg3MzJtQHV1kDCM4u:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L258)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> check number of items in array
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 279 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-Fvj0ct7-8_CeIKU:-KRrg3vyX-m96IZiNf6D:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L279)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> see above comment
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 294 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV-IDIbjyWTE8hg8M2:-KRrg4pmeJJaxU_1LWDm:b8vlyak) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L294)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> see above comment
</details>
Reworked the entire resource. Please review from the beginning again.

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 25 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV1klqmsGtISXbDoyT:-KRrgDjvIIbff23r7b3j:bao32l4) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L25)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> delete these two lines since you don't have anything here
</details>
Reworked the entire test. Please review from the beginning again.

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 43 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV22P4MGQV-AraQ4SY:-KRrgGRVHzQl9TqtCfeO:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L43)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Could you add another test for Get? One for NP probably
</details>
Done

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 99 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV2uxc1cFAJaZp6G1L:-KRrgKiiw8BgexSd7Koh:b-3gigvj) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L99)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> can you re-word this It?
</details>
It's not there anymore in the reworked test

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 128 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV38rE7aRmePTL4ddU:-KRrgbcVXfePwBuV42u-:b-3gigvj) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L128)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> '...is passed an Alias...'
</details>
It's not there anymore in the reworked test

---

*[Tests/Unit/MSFT_xSqlAlias.Tests.ps1, line 132 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV3C7m1L3Sx9OLcLqz:-KRrgcJBpQ9zjyKH93pt:b-3gigvj) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSqlAlias.Tests.ps1#L132)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> see above comment
</details>
It's not there anymore in the reworked test

---

*[Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1, line 106 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV3y-PZJgsNINDnwd5:-KRri33EDggdd3YAHjoI:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1#L106)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> space between ']' and '@'
</details>
Done

---

*[Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1, line 199 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV4f3ymnPcHbczYQKe:-KRri1miOhgi8Xo3vZC5:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1#L199)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> commas should be on the previous line - also space betwen ] and @
</details>
Done

---

*[Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1, line 262 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KRV51dBwQXaMOAaTBq2:-KRrhyxqX2yrhrEprFay:b-6ntyaw) ([raw file](https://github.com/powershell/xsqlserver/blob/7a53ed00b1d9670686f7dc0b6813495a4541c6f6/Tests/Unit/MSFT_xSQLAOGroupEnsure.Tests.ps1#L262)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> space between ] and @
</details>
Done (found a bunch, fixed them all)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-:-KRrdJfpc1PDTvXCGDd2:bluklw1)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 3 of 4 files at r4, 1 of 1 files at r5, 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[README.md, line 333 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3Lp76p3dHx3e8V-P-:-KS3Lp77m9euKY88DPcj:b-cl8313) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/README.md#L333)):*
> ```Markdown
> 
> ### xSqlAlias
>  * **Ensure**: Determines whether the alias should be added or removed. Default values is 'Present'
> ```

'Default value...'

---

*[README.md, line 336 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3M-60WfnyvtkXleAQ:-KS3M-60WfnyvtkXleAR:b-cl8313) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/README.md#L336)):*
> ```Markdown
>  * **Name**: (Key) The name of Alias (e.g. svr01\inst01).
>  * **ServerName**: (Key) The SQL Server you are aliasing (the netbios name or FQDN).
>  * **Protocol**: Protocol to use when connecting. Valid values are 'TCP' or 'NP' (Named Pipes). Default values is 'TCP'.
> ```

'Default value...'

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 230 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RPJyy2eS2uZrqcpW:-KS3RPJzLJl2Es6zVj8y:b6rf2wa) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L230)):*
> ```PowerShell
>     if ($result) 
>     {
>         Write-Verbose "In the desired state"
> ```

single quotes

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 234 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RQQWj3ZOO5VEjqWh:-KS3RQQWj3ZOO5VEjqWi:bucdv7m) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L234)):*
> ```PowerShell
>     else
>     {
>         Write-Verbose "Not in the desired state"
> ```

single quotes - also add in Parameter name '-Message' in this one and the one above

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.schema.mof, line 5 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RZD_Hk2Kxkw12o--:-KS3RZD_Hk2Kxkw12o-0:bvcjyft) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.schema.mof#L5)):*
> ```
> {
>     [Key, Description("The name of Alias (e.g. svr01\\inst01).")] String Name;
>     [Write, Description("Protocol to use when connecting. Valid values are 'TCP' or 'NP' (Named Pipes). Default values is 'TCP'."), ValueMap{"TCP","NP"}, Values{"TCP","NP"}] String Protocol;
> ```

Default 'value'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/114)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[README.md, line 333 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3Lp76p3dHx3e8V-P-:-KS7kYnHWVR-_PTsmiFS:b-xs68wl) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/README.md#L333)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'Default value...'
</details>
Done. Also changed xSqlAlias to xSQLAlias here in the header and in the unreleased section of the README.md.

---

*[README.md, line 336 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3M-60WfnyvtkXleAQ:-KS7katHrJ3qeLzAXftY:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/README.md#L336)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'Default value...'
</details>
Done

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 230 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RPJyy2eS2uZrqcpW:-KS7ke_5V94bTJ9HFYhE:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L230)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> single quotes
</details>
Done

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 234 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RQQWj3ZOO5VEjqWh:-KS7kLtNzX4Ayzc8Im0p:bb6un1j) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L234)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> single quotes - also add in Parameter name '-Message' in this one and the one above
</details>
Done

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.schema.mof, line 5 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RZD_Hk2Kxkw12o--:-KS7kNewh4u-MWXpxPcw:bwwrqid) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.schema.mof#L5)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Default 'value'
</details>
Done. Also found one in the Ensure parameter.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/114)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 234 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RQQWj3ZOO5VEjqWh:-KS7oJ63GJcVIXhB6dDw:bi1srq2) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L234)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Still need to add -Message parameter

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/114)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: 7 of 8 files reviewed at latest revision, 15 unresolved discussions.

---

*[DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1, line 234 at r6](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-KS3RQQWj3ZOO5VEjqWh:-KS7pOAZ9XAerhgVubfU:b-jllz0z) ([raw file](https://github.com/powershell/xsqlserver/blob/e5a945b02116694d63e929a2190b938b8584aa5c/DSCResources/MSFT_xSQLAlias/MSFT_xSQLAlias.psm1#L234)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Still need to add -Message parameter
</details>
Ahh, my bad. I read that before changing the code but forgot to change it. And in reviewable I just checked the code so it was okay and typed 'Done' without reading your comment again. Will not happen again. One learns by ones mistake :smile: 

Really done now! :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/114)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/114#-:-KS8J1IC_fubicfKjaTE:bnfp4nl)*
<!-- Sent from Reviewable.io -->
